### ISSUE TITLE
Qlot install fails if some asd file has defsystem-depends-on from prove-asdf

### ISSUE THREAD

**Issue by svetlyak40wt** [fukamachi/qlot#84]
Here is an example of such system:

https://github.com/takagi/lake/blob/master/lake-test.asd

Full traceback:

```
[art@art-osx:~/projects/lisp/lake]% qlot install
Installing Quicklisp to /Users/art/projects/lisp/lake/.qlot/ ...
; Fetching #<URL "http://beta.quicklisp.org/quicklisp.lisp">
; 55.80KB
Reading '/Users/art/projects/lisp/lake/qlfile'...
; Fetching #<URL "http://beta.quicklisp.org/dist/quicklisp.txt">
; 0.40KB
==================================================
408 bytes in 0.00 seconds (0.00KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/dist/quicklisp.txt">
; 0.40KB
==================================================
408 bytes in 0.00 seconds (0.00KB/sec)
Already have dist "quicklisp" version "2019-08-13".
Calculating project dependencies...
While evaluating the form starting at line 11, column 0
  of #P"/Users/art/projects/lisp/lake/lake-test.asd":
Unhandled TYPE-ERROR in thread #<SB-THREAD:THREAD "main thread" RUNNING
                                  {10004F84C3}>:
  The value
    NIL
  is not of type
    FUNCTION

Backtrace for: #<SB-THREAD:THREAD "main thread" RUNNING {10004F84C3}>
0: ((FLET SB-THREAD::WITH-RECURSIVE-LOCK-THUNK :IN SB-PCL::INSTALL-OPTIMIZED-CONSTRUCTOR))
1: ((FLET "WITHOUT-INTERRUPTS-BODY-29" :IN SB-THREAD::CALL-WITH-RECURSIVE-LOCK))
2: (SB-THREAD::CALL-WITH-RECURSIVE-LOCK #<CLOSURE (FLET SB-THREAD::WITH-RECURSIVE-LOCK-THUNK :IN SB-PCL::INSTALL-OPTIMIZED-CONSTRUCTOR) {11FDA1B}> #<SB-THREAD:MUTEX "World Lock" owner: #<SB-THREAD:THREAD "main thread" RUNNING {10004F84C3}>> T NIL)
3: (SB-PCL::INSTALL-OPTIMIZED-CONSTRUCTOR #<SB-PCL::CTOR QL-CDB::CDB-WRITER (:OUTPUT SB-PCL::|.P0.|)>)
4: ((LAMBDA (&REST SB-PCL::ARGS) :IN SB-PCL::INSTALL-INITIAL-CONSTRUCTOR) #<SB-SYS:FD-STREAM for "file /Users/art/projects/lisp/lake/.qlot/dists/quicklisp/systems.cdb-tmp" {1001614583}>)
5: (QL-CDB::CALL-WITH-OUTPUT-TO-CDB #P"/Users/art/projects/lisp/lake/.qlot/dists/quicklisp/systems.cdb" #P"/Users/art/projects/lisp/lake/.qlot/dists/quicklisp/systems.cdb-tmp" #<CLOSURE (LAMBDA (QL-CDB::CDB) :IN QL-CDB:CONVERT-INDEX-FILE) {1001613A3B}>)
6: (QL-CDB:CONVERT-INDEX-FILE #P"/Users/art/projects/lisp/lake/.qlot/dists/quicklisp/systems.txt" :CDB-FILE #P"/Users/art/projects/lisp/lake/.qlot/dists/quicklisp/systems.cdb" :INDEX 2)
7: ((:METHOD QL-DIST:FIND-SYSTEM-IN-DIST (T QL-DIST:DIST)) "prove-asdf" #<QL-DIST:DIST quicklisp 2019-08-13>) [fast-method]
8: (QL-DIST::FIND-THING-NAMED #<STANDARD-GENERIC-FUNCTION QL-DIST:FIND-SYSTEM-IN-DIST (1)> "prove-asdf")
9: ((:METHOD QL-DIST:FIND-SYSTEM (T)) :PROVE-ASDF) [fast-method]
10: ((LABELS QLOT/SOURCE::GET-DEPS :IN QLOT/SOURCE::MAKE-HOOK) :PROVE-ASDF)
11: ((LAMBDA (QLOT/SOURCE::FUN QLOT/SOURCE::FORM QLOT/SOURCE::ENV) :IN QLOT/SOURCE::MAKE-HOOK) #<FUNCTION (MACRO-FUNCTION DEFSYSTEM) {22629CFB}> (DEFSYSTEM LAKE-TEST :AUTHOR "Rudolph Miller and Masayuki Takagi" :LICENSE "MIT" :HOMEPAGE "https://github.com/takagi/lake" :DEPENDS-ON (:LAKE :PROVE) :COMPONENTS ((:MODULE "t" :COMPONENTS ((:TEST-FILE "lake")))) ...) #<NULL-LEXENV>)
12: ((FLET SB-IMPL::PERFORM-EXPANSION :IN MACROEXPAND-1) #<FUNCTION (MACRO-FUNCTION DEFSYSTEM) {22629CFB}> NIL)
13: (MACROEXPAND (DEFSYSTEM LAKE-TEST :AUTHOR "Rudolph Miller and Masayuki Takagi" :LICENSE "MIT" :HOMEPAGE "https://github.com/takagi/lake" :DEPENDS-ON (:LAKE :PROVE) :COMPONENTS ((:MODULE "t" :COMPONENTS ((:TEST-FILE "lake")))) ...) #<NULL-LEXENV>)
14: (SB-INT:SIMPLE-EVAL-IN-LEXENV (DEFSYSTEM LAKE-TEST :AUTHOR "Rudolph Miller and Masayuki Takagi" :LICENSE "MIT" :HOMEPAGE "https://github.com/takagi/lake" :DEPENDS-ON (:LAKE :PROVE) :COMPONENTS ((:MODULE "t" :COMPONENTS ((:TEST-FILE "lake")))) ...) #<NULL-LEXENV>)
15: (SB-EXT:EVAL-TLF (DEFSYSTEM LAKE-TEST :AUTHOR "Rudolph Miller and Masayuki Takagi" :LICENSE "MIT" :HOMEPAGE "https://github.com/takagi/lake" :DEPENDS-ON (:LAKE :PROVE) :COMPONENTS ((:MODULE "t" :COMPONENTS ((:TEST-FILE "lake")))) ...) 3 NIL)
16: ((LABELS SB-FASL::EVAL-FORM :IN SB-INT:LOAD-AS-SOURCE) (DEFSYSTEM LAKE-TEST :AUTHOR "Rudolph Miller and Masayuki Takagi" :LICENSE "MIT" :HOMEPAGE "https://github.com/takagi/lake" :DEPENDS-ON (:LAKE :PROVE) :COMPONENTS ((:MODULE "t" :COMPONENTS ((:TEST-FILE "lake")))) ...) 3)
17: ((LAMBDA (SB-KERNEL:FORM &KEY :CURRENT-INDEX &ALLOW-OTHER-KEYS) :IN SB-INT:LOAD-AS-SOURCE) (DEFSYSTEM LAKE-TEST :AUTHOR "Rudolph Miller and Masayuki Takagi" :LICENSE "MIT" :HOMEPAGE "https://github.com/takagi/lake" :DEPENDS-ON (:LAKE :PROVE) :COMPONENTS ((:MODULE "t" :COMPONENTS ((:TEST-FILE "lake")))) ...) :CURRENT-INDEX 3)
18: (SB-C::%DO-FORMS-FROM-INFO #<CLOSURE (LAMBDA (SB-KERNEL:FORM &KEY :CURRENT-INDEX &ALLOW-OTHER-KEYS) :IN SB-INT:LOAD-AS-SOURCE) {100160707B}> #<SB-C::SOURCE-INFO {1001607033}> SB-C::INPUT-ERROR-IN-LOAD)
19: (SB-INT:LOAD-AS-SOURCE #<SB-INT:FORM-TRACKING-STREAM for "file /Users/art/projects/lisp/lake/lake-test.asd" {1001605343}> :VERBOSE NIL :PRINT NIL :CONTEXT "loading")
20: ((FLET SB-FASL::THUNK :IN LOAD))
21: (SB-FASL::CALL-WITH-LOAD-BINDINGS #<CLOSURE (FLET SB-FASL::THUNK :IN LOAD) {11FE66B}> #<SB-INT:FORM-TRACKING-STREAM for "file /Users/art/projects/lisp/lake/lake-test.asd" {1001605343}>)
22: ((FLET SB-FASL::LOAD-STREAM :IN LOAD) #<SB-INT:FORM-TRACKING-STREAM for "file /Users/art/projects/lisp/lake/lake-test.asd" {1001605343}> NIL)
23: (LOAD #P"/Users/art/projects/lisp/lake/lake-test.asd" :VERBOSE NIL :PRINT NIL :IF-DOES-NOT-EXIST T :EXTERNAL-FORMAT :DEFAULT)
24: (QLOT/SOURCE::SYSTEM-FILE-SYSTEMS #P"/Users/art/projects/lisp/lake/lake-test.asd")
25: ((LAMBDA NIL :IN QLOT/INSTALL::APPLY-QLFILE-TO-QLHOME))
26: (QLOT/UTIL:CALL-IN-LOCAL-QUICKLISP #<CLOSURE (LAMBDA NIL :IN QLOT/INSTALL::APPLY-QLFILE-TO-QLHOME) {10015AF17B}> #P"/Users/art/projects/lisp/lake/.qlot/" :SYSTEMS (#P"/Users/art/projects/lisp/lake/lake.asd" #P"/Users/art/projects/lisp/lake/lake-test.asd") :CENTRAL-REGISTRY (#P"/Users/art/.roswell/local-projects/fukamachi/qlot/"))
27: (QLOT/INSTALL::APPLY-QLFILE-TO-QLHOME #P"/Users/art/projects/lisp/lake/qlfile" #P"/Users/art/projects/lisp/lake/.qlot/" :IGNORE-LOCK NIL :PROJECTS NIL)
28: (QLOT/INSTALL:INSTALL-QLFILE #P"/Users/art/projects/lisp/lake/qlfile" :QUICKLISP-HOME #P"/Users/art/projects/lisp/lake/.qlot/")
29: (ROS/SCRIPT/QLOT::MAIN "install")
30: (SB-INT:SIMPLE-EVAL-IN-LEXENV (APPLY (QUOTE ROS/SCRIPT/QLOT::MAIN) ROSWELL:*ARGV*) #<NULL-LEXENV>)
31: (SB-INT:SIMPLE-EVAL-IN-LEXENV (ROSWELL:QUIT (APPLY (QUOTE ROS/SCRIPT/QLOT::MAIN) ROSWELL:*ARGV*)) #<NULL-LEXENV>)
32: (SB-EXT:EVAL-TLF (ROSWELL:QUIT (APPLY (QUOTE ROS/SCRIPT/QLOT::MAIN) ROSWELL:*ARGV*)) NIL NIL)
33: ((LABELS SB-FASL::EVAL-FORM :IN SB-INT:LOAD-AS-SOURCE) (ROSWELL:QUIT (APPLY (QUOTE ROS/SCRIPT/QLOT::MAIN) ROSWELL:*ARGV*)) NIL)
34: (SB-INT:LOAD-AS-SOURCE #<CONCATENATED-STREAM :STREAMS NIL {1002D0BCA3}> :VERBOSE NIL :PRINT NIL :CONTEXT "loading")
35: ((FLET SB-FASL::THUNK :IN LOAD))
36: (SB-FASL::CALL-WITH-LOAD-BINDINGS #<CLOSURE (FLET SB-FASL::THUNK :IN LOAD) {11FF53B}> #<CONCATENATED-STREAM :STREAMS NIL {1002D0BCA3}>)
37: ((FLET SB-FASL::LOAD-STREAM :IN LOAD) #<CONCATENATED-STREAM :STREAMS NIL {1002D0BCA3}> NIL)
38: (LOAD #<CONCATENATED-STREAM :STREAMS NIL {1002D0BCA3}> :VERBOSE NIL :PRINT NIL :IF-DOES-NOT-EXIST T :EXTERNAL-FORMAT :DEFAULT)
39: ((FLET ROSWELL::BODY :IN ROSWELL:SCRIPT) #<SB-SYS:FD-STREAM for "file /Users/art/.roswell/bin/qlot" {1002D06A63}>)
40: (ROSWELL:SCRIPT "/Users/art/.roswell/bin/qlot" "install")
41: (ROSWELL:RUN ((:SCRIPT "/Users/art/.roswell/bin/qlot" "install") (:QUIT NIL)))
42: (SB-INT:SIMPLE-EVAL-IN-LEXENV (ROSWELL:RUN (QUOTE ((:SCRIPT "/Users/art/.roswell/bin/qlot" "install") (:QUIT NIL)))) #<NULL-LEXENV>)
43: (EVAL (ROSWELL:RUN (QUOTE ((:SCRIPT "/Users/art/.roswell/bin/qlot" "install") (:QUIT NIL)))))
44: (SB-IMPL::PROCESS-EVAL/LOAD-OPTIONS ((:EVAL . "(progn #-ros.init(cl:load \"/Users/art/.local/etc/roswell/init.lisp\"))") (:EVAL . "(ros:run '((:script \"/Users/art/.roswell/bin/qlot\"\"install\")(:quit ())))")))
45: (SB-IMPL::TOPLEVEL-INIT)
46: ((FLET SB-UNIX::BODY :IN SB-EXT:SAVE-LISP-AND-DIE))
47: ((FLET "WITHOUT-INTERRUPTS-BODY-14" :IN SB-EXT:SAVE-LISP-AND-DIE))
48: ((LABELS SB-IMPL::RESTART-LISP :IN SB-EXT:SAVE-LISP-AND-DIE))

unhandled condition in --disable-debugger mode, quitting
```

`qlfile` is empty

### Workaround

After this error, do:

```
qlot exec ros run --eval '(ql:quickload :prove-asdf)'  --eval '(uiop:quit)'
```

and run `qlot install again.

### Environment

OSX
Qlot's version: 0.9.13 (68737cf3657a40f2ea88fc0fce3c7c53e0cf2f97 commit)

**Comment by fukamachi** [comment]
Confirmed and fixed at #85. I'll merge it when its CI passed.

**Comment by svetlyak40wt** [comment]
Great! Thank you!
